### PR TITLE
Spring Boot / Spring Boot Practice / Add Order model

### PR DESCRIPTION
**Pull request by IvanovYurii**
1. Refactor code:
    - Rewrite updateBook method in BookServiceImpl. When the fields are updated, they can be null. The isbn field is checked for uniqueness
    - Private methods are moved before public
    - The Shopping Cart creation method has been moved to UserService. The Shopping Cart  is created automatically when the user registers.
    - Rewritten for Swagger all descriptions in controllers
    - Added annotation **@ Min()**  to requests for validation Long type.
2. Solution of this task:
   - Added entities:
      - Order
      - OrderItems.
   - Added scripts for creation tables:
      - orders
      - order_items
   - Added for the entities:
      - DTOs
      - mappers
      - repositories
      - services
   - Added OrderController with Implemented Order endpoints:
      - Place an order
      - Retrieve user's order history
      - Update order status
      - Retrieve all OrderItems for a specific order
      - Retrieve a specific OrderItem within an order

### CODE REVIEW

File: src/main/java/ivanov/springbootintro/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -17,15 +19,24 @@ public class UserServiceImpl implements UserService {
...
+
+    void registerNewShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+    }
 
     @Override
-    public UserResponseDto save(UserRegistrationRequestDto requestDto)
+    public UserResponseDto registerUser(UserRegistrationRequestDto requestDto)
```
Comment (okuzan) [inline src/main/java/ivanov/springbootintro/service/impl/UserServiceImpl.java]:
```suggestion
    public UserResponseDto register(UserRegistrationRequestDto requestDto)

```

File: src/main/java/ivanov/springbootintro/service/impl/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        orderRepository.save(order);
+        return orderMapper.orderToOrderDto(order);
+    }
+
+    @Transactional
+    @Override
+    public List<OrderItemDto> getOrderItems(Long orderId, Pageable pageable) {
+        Order order = orderRepository.findById(orderId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find order by id=" + orderId)
+        );
```
Comment (okuzan) [inline src/main/java/ivanov/springbootintro/service/impl/OrderServiceImpl.java]: repeated logic, encapsulate in a separate method

File: src/main/java/ivanov/springbootintro/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    @Transactional
+    @Override
+    public OrderDto updateOrderStatus(Long orderId, UpdateStatusOrderRequestDto request) {
+        Order order = orderRepository.findById(orderId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find order by id=" + orderId)
+        );
+        Order.Status newStatus = Order.Status.fromValue(request.status()).orElseThrow(
+                () -> new EntityNotFoundException("Invalid status: " + request.status()));
```
Comment (okuzan) [inline src/main/java/ivanov/springbootintro/service/impl/OrderServiceImpl.java]: encapsulate in a separate method

File: src/main/java/ivanov/springbootintro/service/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                    orderItem.setQuantity(cartItem.getQuantity());
+                    orderItem.setPrice(cartItem.getBook().getPrice());
+                    return orderItem;
+                })
+                .collect(Collectors.toSet());
+    }
+
+    private BigDecimal calculateTotal(Set<OrderItem> orderItems) {
+        return orderItems.stream()
+                .map(item -> item.getPrice().multiply(BigDecimal.valueOf(item.getQuantity())))
```
Comment (okuzan) [inline src/main/java/ivanov/springbootintro/service/impl/OrderServiceImpl.java]: place private method after public ones

File: src/main/java/ivanov/springbootintro/service/BookService.java [hunk 5]
```diff
@@ -7,15 +7,15 @@
 import org.springframework.data.domain.Pageable;
 
 public interface BookService {
```
Comment (okuzan) [inline src/main/java/ivanov/springbootintro/service/BookService.java:9]: previous naming was more common TBH, it's bookService, we don't need to mention "book" in every method

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/ivanov/springbootintro/service/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+        return cartItems.stream()
+                .map(cartItem -> {
+                    OrderItem orderItem = new OrderItem();
+                    orderItem.setOrder(order);
+                    orderItem.setBook(cartItem.getBook());
+                    orderItem.setQuantity(cartItem.getQuantity());
+                    orderItem.setPrice(cartItem.getBook().getPrice());
+                    return orderItem;
+                })
+                .collect(Collectors.toSet());
```
Comment (okuzan) [inline src/main/java/ivanov/springbootintro/service/impl/OrderServiceImpl.java]: you can consider refactoring to this if you want to 
```
private Set<OrderItem> createOrderItems(Order order, Set<CartItem> cartItems) {
    return cartItems.stream()
            .map(cartItem -> createOrderItemFromCartItem(order, cartItem))
            .collect(Collectors.toSet());
}

private OrderItem createOrderItemFromCartItem(Order order, CartItem cartItem) {
    OrderItem orderItem = new OrderItem();
    orderItem.setOrder(order);
    orderItem.setBook(cartItem.getBook());
    orderItem.setQuantity(cartItem.getQuantity());
    orderItem.setPrice(cartItem.getBook().getPrice());
    return orderItem;
}

```

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**